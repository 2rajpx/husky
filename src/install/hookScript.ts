// Used to identify scripts created by Husky
export const huskyIdentifier = '# husky'

export default `#!/bin/sh
${huskyIdentifier}
pwd
./node_modules/husky/scripts/run.sh
`
