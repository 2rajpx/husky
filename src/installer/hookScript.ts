import * as fs from 'fs'
import * as path from 'path'

// Used to identify scripts created by Husky
export const huskyIdentifier = '# husky'

export default fs
  .readFileSync(path.join(__dirname, '../../templates/hook.sh'), 'utf-8')
  .replace('__huskyIdentifier__', huskyIdentifier)
