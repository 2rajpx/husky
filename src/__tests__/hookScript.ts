import hookScript from '../install/hookScript'

describe('hookScript', () => {
  it('should match snapshot', () => {
    expect(hookScript).toMatchSnapshot()
  })
})
