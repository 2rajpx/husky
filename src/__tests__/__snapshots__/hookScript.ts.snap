// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hookScript should match snapshot 1`] = `
"#!/bin/sh
# husky

hookname=\\\\\`basename \\"$0\\"\\\\\`
[ -f package.json ] && cat package.json | grep -q \\"\\\\\\\\\\"$hookname\\\\\\\\\\"[[:space:]]*:\\"

if [[ $? -eq 0 ]]; then
  ./node_modules/.bin/run-node ./node_modules/husky/lib/run $hookname
  exit $?
fi"
`;
